test/src/test/java/hudson/diagnosis/TooManyJobsButNoViewTest.java

```diff
@@ -67,4 +80,57 @@ private void verifyNoForm() throws IOException, SAXException {
 
         verifyNoForm();
     }
+    
+    @Test
+    public void systemReadNoViewAccessVerifyNoForm() throws Exception {
+        final String READONLY = "readonly";
+
+        r.jenkins.setSecurityRealm(r.createDummySecurityRealm());
+        r.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy()
+                .grant(Jenkins.READ).everywhere().to(READONLY)
+                .grant(Jenkins.SYSTEM_READ).everywhere().to(READONLY)
+        );
+
+        for (int i = 0; i <= TooManyJobsButNoView.THRESHOLD; i++)
+            r.createFreeStyleProject();
+
+        JenkinsRule.WebClient wc = r.createWebClient();
+        wc.login(READONLY);
+
+        verifyNoMonitor(wc);
+    }
+
+    private void verifyNoMonitor(JenkinsRule.WebClient wc) throws IOException, SAXException {
+        HtmlPage p = wc.goTo("manage");
+        DomElement adminMonitorDiv = p.getElementById("tooManyJobsButNoView");
+        assertThat(adminMonitorDiv, is(nullValue()));
+    }
+    
+    @Test
+    public void systemReadVerifyForm() throws Exception {
+        final String READONLY = "readonly";
+
+        r.jenkins.setSecurityRealm(r.createDummySecurityRealm());
+        r.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy()
+                .grant(Jenkins.READ).everywhere().to(READONLY)
+                .grant(Jenkins.SYSTEM_READ).everywhere().to(READONLY)
+                .grant(Item.READ).everywhere().to(READONLY)
+                .grant(View.READ).everywhere().to(READONLY)
+        );
+
+        for (int i = 0; i <= TooManyJobsButNoView.THRESHOLD; i++)
+            r.createFreeStyleProject();
+
+        JenkinsRule.WebClient wc = r.createWebClient();
+        wc.login(READONLY);
+
+        verifyMonitor(wc);
+    }
+
+    private void verifyMonitor(JenkinsRule.WebClient wc) throws IOException, SAXException {
+        HtmlPage p = wc.goTo("manage");
+        DomElement adminMonitorDiv = p.getElementById("tooManyJobsButNoView");
+        assertThat(adminMonitorDiv.getTextContent(), containsString("There appears to be a large number of jobs"));
```

Comment:
Potential i18n problem? I'd be happy with confirming the element exists.

Reply:
sure, was just an extra check for safety, I'll simplify